ParkPlanner/Pages/Index.cshtml.cs

```diff
@@ -8,26 +8,42 @@ namespace ParkPlanner.Pages
     public class IndexModel : PageModel
     {
         private readonly ILogger<IndexModel> _logger;
+        private readonly ConfigurationResolver _configurationResolver;
         static readonly HttpClient client = new HttpClient();
-        public IndexModel(ILogger<IndexModel> logger)
+        public IndexModel(ILogger<IndexModel> logger,
+            ConfigurationResolver configurationResolver)
         {
             _logger = logger;
+            _configurationResolver = configurationResolver;
         }
 
         public void OnGet()
         {
+            
+            try
+            {
+                var URL = _configurationResolver.GetConfiguration("npsURL");
+                var task = client.GetAsync(URL);
+                HttpResponseMessage result = task.Result;
+                List<Datum> park = new List<Datum>();
+                if (result.IsSuccessStatusCode)
+                {
+                    Task<string> readString = result.Content.ReadAsStringAsync();
+                    string jsonString = readString.Result;
+                    park = Park.FromJson(jsonString).Data;
 
-            var task = client.GetAsync("https://developer.nps.gov/api/v1/parks?api_key=mLBONbm3NZfawfBoS0w4bXT3yyJ1nBpLhqh6o0Au\r\n");
-            HttpResponseMessage result = task.Result;
-            List<Datum> park = new List<Datum>();
-            if (result.IsSuccessStatusCode)
+                }
+                else
+                {
+                    _logger.LogInformation("Http Request Failure", result.StatusCode);
+                }
+                ViewData["Park"] = park;
+            }
+            catch (Exception)
             {
-                Task<string> readString = result.Content.ReadAsStringAsync();
-                string jsonString = readString.Result;
-                park = Park.FromJson(jsonString).Data;
 
+                ViewData["ErrorMessage"] = "Http Call and JSON Parsing resulted in an Exception";
```

Comment:
Consider adding a logging statement here.
Exception handling is generally a good idea, though consider a few ways to make it more effective:
- Heal the exception, if at all possible.  That's the ultimate goa.
- Log the exception, using proper logging.
- If not healed, how will you inform the calling method that an exception occurred?  Re throw the exception?  Return an empty collection?